File: java/bubbleSort.java

import java.util.Scanner;
public class bubbleSort {
    public void bubble(int[] arr,int n){
       for(int i=0;i<n-1;i++){
        for(int j=0;j<n-i-1;j++){
            if(arr[j]>arr[j+1]){
                int temp=arr[j];
                arr[j]=arr[j+1];
                arr[j+1]=temp;
            }
        }
       }
    }
    public static void main(String[] args) {
        Scanner sc=new Scanner(System.in);
        System.out.println("enter the size of array:");
        int n=sc.nextInt();
        int[] arr=new int[n];
        System.out.println("enter the elements of array:");
        for(int i=0;i<n;i++){
            arr[i]=sc.nextInt();
        }
        bubbleSort sorter=new bubbleSort();
        sorter.bubble(arr,n);
        for(int j=0;j<n;j++){
           System.out.println(arr[j]);
        }
        sc.close();
    }
}

File: C/bubbleSort.c
#include<stdio.h>
void printarray(int *a ,int n)
{
	int i;
	for(i=0;i<n;i++)
	{
		printf("%d\n",a[i]);
	}
	printf("\n");
}
void bubblesort(int *a,int n){
	int t,i,j;
	for(i=0;i<n-1;i++)
	{
		for(j=0;j<n-1-i;j++)
		{
			if(a[j]>a[j+1])
			{
				t=a[j];
				a[j]=a[j+1];
				a[j+1]=t;
			}
		}
	}
}
int main()
{
	int n=5;
	int a[5],i;
	printf("enter the elements of array:");
	for(i=0;i<5;i++)
	{
		scanf("%d",&a[i]);
	}
	printarray(a,n);
	bubblesort(a,n);
	printarray(a,n);
	return 0;
}
	  
